sbi/user_input/user_input_checks.py

```diff
@@ -399,32 +399,24 @@ def wrap_as_batch_simulator(simulator: Callable, prior) -> Callable:
         assert simulator_batch_size == batch_size
     except:
         is_batched_simulator = False
-        warnings.warn(
-            """Simulator can't handle batches of parameters. It will loop over batches
-            in Python with `map`. Consider vectorising the simulator natively for
-            performance."""
-        )
-        # make sure the simulator does not return batch dim.
-        single_simulation = simulator(prior.sample())
-        assert (
-            not single_simulation.shape[0] == 1
-        ), f"""The simulator can't handle batches and returns a singleton batch
-            dimension: {single_simulation.shape}. A simulator that can't handle batches
-            of parameters must not return a batch dim."""
 
-    return simulator if is_batched_simulator else get_batched_simulator(simulator)
+    return simulator if is_batched_simulator else get_batch_dim_simulator(simulator)
 
 
-def get_batched_simulator(simulator: Callable) -> Callable:
+def get_batch_dim_simulator(simulator: Callable) -> Callable:
     """Return simulator wrapped with `map` to handle batches of parameters."""
 
-    def batched_simulator(theta: Tensor) -> Tensor:
-        theta = ensure_theta_batched(theta)
-        # XXX: this should be handled with more care, e.g., enable multiprocessing
-        # XXX: with Pool() as p: p.map(...)
-        return torch.stack(list(map(simulator, theta)))
+    def batch_dim_simulator(theta: Tensor) -> Tensor:
+        batch_shape, *_ = theta.shape
+        assert (
+            batch_shape == 1
```

Comment:
Just to make sure I understand it correctly: If the user provides `simulation_batch_size` larger than 1, they will give an assert error here, right?

Reply:
Yes, because their simulator cant handle batches and we agreed to not provide a wrapper for that.